HostLibraryTests/llvm/LibraryPerformance_test.cpp

```diff
@@ -35,58 +34,156 @@
 
 using namespace Tensile;
 
-struct LibraryPerformanceTest: public ::testing::TestWithParam<std::tuple<std::string, bool>>
+/**
+ * LibraryPerformanceTest:
+ * 
+ * This suite contains micro-benchmarks for pieces of the runtime library.  It does not
+ * exercise any of the Hip-specific code.
+ * 
+ * There are no performance-based assertions or checks.  The timing results are provided by
+ * googletest.
+ * 
+ * Most of these tests depend on a library being loaded from a YAML file.  The library objects
+ * are cached so that the deserialization time is not a part of the actual test (outside of the
+ * LoadLibrary test). PopulateCache is an empty test whose purpose is to ensure the cache is
+ * populated for the actual tests.
+ */
+struct LibraryPerformanceTest: public ::testing::TestWithParam<std::tuple<AMDGPU, std::string, bool, bool>>
 {
+    AMDGPU hardware;
+    std::string filename;
+    bool hasNavi, solutionRequired;
+    std::shared_ptr<SolutionLibrary<ContractionProblem>> library;
+
+    static std::map<std::string, std::shared_ptr<SolutionLibrary<ContractionProblem>>> libraryCache;
+
+	void SetUp() override
+    {
+        std::tie(hardware, filename, hasNavi, solutionRequired) = GetParam();
+
+        if(hardware.processor == AMDGPU::Processor::gfx1010 && !hasNavi)
+            GTEST_SKIP();
+
+        library = loadLibrary();
+        ASSERT_NE(library, nullptr) << filename;
+    }
+
+    std::string libraryPath()
+    {
+        return TestData::Instance().file(filename).native();
+    }
+
+    std::shared_ptr<SolutionLibrary<ContractionProblem>> loadLibrary(bool cache = true)
+    {
+        if(!cache)
+            return LoadLibraryFile<ContractionProblem>(libraryPath());
+
+        auto path = libraryPath();
+
+        auto iter = libraryCache.find(path);
+        if(iter != libraryCache.end())
+            return iter->second;
+
+        return libraryCache[path] = LoadLibraryFile<ContractionProblem>(libraryPath());
+    }
 };
 
+std::map<std::string, std::shared_ptr<SolutionLibrary<ContractionProblem>>> LibraryPerformanceTest::libraryCache;
+
+TEST_P(LibraryPerformanceTest, PopulateCache)
+{
```

Comment:
Is this test supposed to be empty?

Reply:
Yes, see the comment above.  I'll add a comment in the body of the test to make it clearer.